postgis/lwgeom_transform.c

```diff
@@ -225,3 +226,215 @@ Datum LWGEOM_asKML(PG_FUNCTION_ARGS)
 		PG_RETURN_TEXT_P(kml);
 	PG_RETURN_NULL();
 }
+
+
+/********************************************************************************
+ * PROJ database reading functions
+ */
+
+
+#if POSTGIS_PROJ_VERSION >= 61
+
+struct srs_entry {
+	text* auth_name;
+	text* auth_code;
+};
+
+struct srs_data {
+	struct srs_entry* entries;
+	uint32_t num_entries;
+	uint32_t capacity;
+	uint32_t current_entry;
+};
+
+static Datum
+srs_tuple_from_entry(const struct srs_entry* entry, TupleDesc tuple_desc)
+{
+	HeapTuple tuple;
+	Datum tuple_data[4] = {0, 0, 0, 0};
+	bool  tuple_null[4] = {false, false, false, false};
+	PJ_CONTEXT *ctx = NULL;
+	const char* const empty_options[2] = {NULL};
+	const char* const wkt_options[2] = {"MULTILINE=NO", NULL};
+	const char *srtext;
+	const char *proj4text;
+
+	PJ *obj = proj_create_from_database(ctx,
+		text_to_cstring(entry->auth_name),
+		text_to_cstring(entry->auth_code),
+		PJ_CATEGORY_CRS, 0, empty_options);
+
+	if (!obj)
+		return (Datum) 0;
+
+	srtext = proj_as_wkt(ctx, obj, PJ_WKT1_GDAL, wkt_options);
+	proj4text = proj_as_proj_string(ctx, obj, PJ_PROJ_4, empty_options);
```

Comment:
I should warn while this will work, newer PROJ versions are much more restrictive on the datum shift information they put (+towgs84 to be clear). Older PROJ versions would select one of the available transformations among many possible. PROJ >= 6 will only put a +towgs84 clause if there's a single Helmert transformation to WGS84. If there are several ones, it will not pick up a "random" one.